src/api/transform/reasoning.ts

```diff
@@ -136,15 +141,15 @@ export const getGeminiReasoning = ({
 		| "disable"
 		| undefined
 
-	// Respect “off” / unset semantics from the effort selector itself.
+	// Respect "off" / unset semantics from the effort selector itself.
 	if (!selectedEffort || selectedEffort === "disable") {
 		return undefined
 	}
 
-	// Effort-based models on Google GenAI currently support only explicit low/high levels.
-	if (selectedEffort !== "low" && selectedEffort !== "high") {
+	// Effort-based models on Google GenAI support minimal/low/medium/high levels.
+	if (!GEMINI_THINKING_LEVELS.includes(selectedEffort as GeminiThinkingLevel)) {
 		return undefined
 	}
```

Comment:
@roomote Let's make this a proper type guard function called `isGeminiThinkingLevel`.